src/ComponentInstantiationTransformer.cpp

```diff
@@ -81,45 +79,25 @@ struct ComponentContent {
         relations.push_back(std::move(rel));
     }
 
-    void add(std::unique_ptr<AstLoad>& directive, ErrorReport& report) {
-        // Check if load directive already exists
-        auto foundItem = std::find_if(loads.begin(), loads.end(), [&](const std::unique_ptr<AstLoad>& load) {
-            return load->getQualifiedName() == directive->getQualifiedName();
+    void add(std::unique_ptr<AstIO>& directive, ErrorReport& report) {
+        // Check if i/o directive already exists
+        auto foundItem = std::find_if(ios.begin(), ios.end(), [&](const std::unique_ptr<AstIO>& io) {
+            return io->getQualifiedName() == directive->getQualifiedName();
         });
         // if yes, add error
-        if (foundItem != loads.end()) {
-            Diagnostic err(Diagnostic::ERROR,
-                    DiagnosticMessage(
-                            "Redefinition of IO directive " + toString(directive->getQualifiedName()),
-                            directive->getSrcLoc()),
-                    {DiagnosticMessage("Previous definition", (*foundItem)->getSrcLoc())});
-            report.addDiagnostic(err);
-        }
-        // if not, add it
-        loads.push_back(std::move(directive));
-    }
-
-    void add(std::unique_ptr<AstPrintSize>& directive, ErrorReport& report) {
-        // Check if load directive already exists
-        auto foundItem = std::find_if(
-                printSizes.begin(), printSizes.end(), [&](const std::unique_ptr<AstPrintSize>& printSize) {
-                    return printSize->getQualifiedName() == directive->getQualifiedName();
-                });
-        // if yes, add error
-        if (foundItem != printSizes.end()) {
-            Diagnostic err(Diagnostic::ERROR,
-                    DiagnosticMessage(
-                            "Redefinition of IO directive " + toString(directive->getQualifiedName()),
-                            directive->getSrcLoc()),
-                    {DiagnosticMessage("Previous definition", (*foundItem)->getSrcLoc())});
-            report.addDiagnostic(err);
+        if (foundItem != ios.end()) {
+            const std::string& op = (*foundItem)->getKVP("operation");
```

Comment:
The previous code was actually slightly buggy. Multiple AstLoads are okay, allowing loading data from multiple files.

I would actually be okay with multiple printsizes, too. A change in behaviour since we don't currently allow it, and fairly pointless, but it's not really an error.

Reply:
Let's open an issue about this issue.